### PR TITLE
Implement singleton pattern for ComWrappers

### PR DESCRIPTION
**Pull request by j0shuams**
Make ComWrappers robust/have an idempotent setter via singleton pattern. 

Addresses #623

### CODE REVIEW

**Review by Scottj1s — APPROVED** [review]
perfect!

File: src/WinRT.Runtime/ComWrappersSupport.net5.cs [hunk 1]
```diff
@@ -165,6 +169,23 @@ public static void InitializeComWrappers(ComWrappers wrappers = null)
 
     public class DefaultComWrappers : ComWrappers
     {
+        // Instance field and property for Singleton pattern: ComWrappers setter method should be idempotent 
+        private static DefaultComWrappers _instance;
+        public static DefaultComWrappers Instance
```
Comment (manodasanW) [inline src/WinRT.Runtime/ComWrappersSupport.net5.cs:174]: Looks like a new public API.  Probably want to make it internal.

File: src/WinRT.Runtime/ComWrappersSupport.net5.cs [hunk 2]
```diff
@@ -15,6 +15,20 @@ namespace WinRT
 {
     public static partial class ComWrappersSupport
     {
+        // Instance field and property for Singleton pattern: ComWrappers `set` method should be idempotent 
+        private static DefaultComWrappers _instance;
+        internal static DefaultComWrappers Instance
```
Comment (manodasanW) [inline src/WinRT.Runtime/ComWrappersSupport.net5.cs:20]: for clarity, consider naming it DefaultComWrappersInstance.

Comment (manodasanW) [inline src/WinRT.Runtime/ComWrappersSupport.net5.cs:20]: Given we moved it here, probably make it private too.

File: src/WinRT.Runtime/ComWrappersSupport.net5.cs [hunk 3]
```diff
@@ -30,7 +44,7 @@ private static ComWrappers ComWrappers
                     {
                         if (_comWrappers is null)
                         {
-                            _comWrappers = new DefaultComWrappers();
+                            _comWrappers = Instance;
```
Comment (manodasanW) [inline src/WinRT.Runtime/ComWrappersSupport.net5.cs:47]: In both cases, I think you want to do the register first and then set the _comWrappers variable.  As I see for a potential race where if the get is called twice in somewhat parallel to each other, we can end up with one getting the comwrappers instance before it has been registered.

**Review by manodasanW — APPROVED** [review]

**Pull request merged**